packages/mui-material/src/Popover/Popover.d.ts

```diff
@@ -107,7 +107,7 @@ export interface PopoverProps
    *
    * @default {}
    */
-  PaperProps?: Partial<PaperProps>;
+  PaperProps?: Partial<PaperProps<React.ElementType>>;
```

Comment:
What are these changes for?:

```diff
-PaperProps?: Partial<PaperProps>;
+PaperProps?: Partial<PaperProps<React.ElementType>>;
```

Reply:
PaperProps is converted to generic to support `component` type when used within popover. For example below code won't error right now

```
const paperProps: PaperProps<'span'> = {
  component: 'span',
  onClick: (event) => {
    expectType<React.MouseEvent<HTMLSpanElement, MouseEvent>, typeof event>(event);
  },
};
function Test() {
  return <Popover open PaperProps={paperProps} />;
}
```

Reply:
@DiegoAndai I'll try to explain. In this PR, `PaperProps` is not specifically converted to a generic type. It was already a generic type. With the addition of the `component` type to `OverrideProps` (which `PaperProps` uses), it can now support the component prop. To enable support for elements other than the default `<div>` (default component of Paper), we need to provide `React.ElementType` here.

@sai6855, I noticed that `PaperProps` is deprecated, and we recommend using `slotProps.paper` instead. However, the `component` type is missing there. Could you check if it's easy to add? If not, we can handle it in a separate PR

Reply:
> If not, we can handle it in a separate PR

I think it's better to do this change in follow up PR, as this PR already became too big. I'll take ownership for this change